stl/inc/xutility

```diff
@@ -5504,9 +5504,75 @@ _NODISCARD constexpr _Iter_diff_t<_InIt> _Count_pr(_InIt _First, const _InIt _La
     return _Count;
 }
 
+enum class _TrimResult : unsigned char { _Continue, _Break, _ReturnFalse, _ReturnTrue };
+
+template <class _BidIt1, class _BidIt2, class _Pr>
+_NODISCARD _CONSTEXPR20 _TrimResult _Trim_equal(
+    _BidIt1& _First1, _BidIt1& _Last1, _BidIt2& _First2, _BidIt2& _Last2, _Pr _Pred) {
+    _STL_INTERNAL_CHECK(_First1 != _Last1);
+    _STL_INTERNAL_CHECK(_STD distance(_First1, _Last1) == _STD distance(_First2, _Last2));
+    if (_Pred(*_First1, *_First2)) {
+        do {
+            ++_First1;
+            ++_First2;
+            if (_First1 == _Last1) {
+                // only one element is left
+                return _Pred(*_First1, *_First2) ? _TrimResult::_ReturnTrue : _TrimResult::_ReturnFalse;
+            }
+        } while (_Pred(*_First1, *_First2));
+    } else {

```

Comment:
I mgiht be missing something but this does 
```cpp
if (meow) {
    do {} while(meow)
}
```
Which is equivalent to 
```cpp
while(meow) {}
```

Reply:
You're missing the `else` block. We invoke the `else` block only if the first comparison fails.